File: backend/src/main/java/com/smartpark/service/VehicleService.java
package com.smartpark.service;

import com.smartpark.dto.request.VehicleRequest;
import com.smartpark.dto.response.VehicleResponse;

import java.util.List;

public interface VehicleService {
    VehicleResponse addVehicle(Long userId, VehicleRequest request);
    List<VehicleResponse> getUserVehicles(Long userId);
    VehicleResponse getVehicleById(Long vehicleId, Long userId);
    VehicleResponse updateVehicle(Long vehicleId, Long userId, VehicleRequest request);
    void deleteVehicle(Long vehicleId, Long userId);
}

File: frontend/src/services/vehicleService.js
import api from './api';

export const vehicleService = {
  getMyVehicles: async () => {
    return api.get('/vehicles');
  },

  getVehicleById: async (id) => {
    return api.get(`/vehicles/${id}`);
  },

  addVehicle: async (data) => {
    return api.post('/vehicles', data);
  },

  updateVehicle: async (id, data) => {
    return api.put(`/vehicles/${id}`, data);
  },

  deleteVehicle: async (id) => {
    return api.delete(`/vehicles/${id}`);
  }
};
